plasma_framework/contracts/src/exits/payment/controllers/PaymentChallengeIFEOutput.sol

```diff
@@ -0,0 +1,168 @@
+pragma experimental ABIEncoderV2;
+
+import "../PaymentExitDataModel.sol";
+import "../PaymentInFlightExitModelUtils.sol";
+import "../routers/PaymentInFlightExitRouterArgs.sol";
+import "../spendingConditions/IPaymentSpendingCondition.sol";
+import "../spendingConditions/PaymentSpendingConditionRegistry.sol";
+import "../../models/OutputGuardModel.sol";
+import "../../interfaces/IOutputGuardHandler.sol";
+import "../../registries/OutputGuardHandlerRegistry.sol";
+import "../../utils/ExitId.sol";
+import "../../../utils/UtxoPosLib.sol";
+import "../../../utils/Merkle.sol";
+import "../../../transactions/WireTransaction.sol";
+import "../../../framework/PlasmaFramework.sol";
+import "../../../transactions/PaymentTransactionModel.sol";
+import "../../utils/TxFinalization.sol";
+
+library PaymentChallengeIFEOutput {
+    using UtxoPosLib for UtxoPosLib.UtxoPos;
+
+    uint256 public constant PIGGYBACK_BOND = 31415926535 wei;
+
+    struct Controller {
+        PlasmaFramework framework;
+        PaymentSpendingConditionRegistry spendingConditionRegistry;
+        OutputGuardHandlerRegistry outputGuardHandlerRegistry;
+    }
+
+    event InFlightExitOutputBlocked(
+        address indexed challenger,
+        bytes32 ifeTxHash,
+        uint16 outputIndex
+    );
+
+    function buildController(
+        PlasmaFramework framework,
+        PaymentSpendingConditionRegistry spendingConditionRegistry,
+        OutputGuardHandlerRegistry outputGuardHandlerRegistry
+    )
+        public
+        pure
+        returns (Controller memory)
+    {
+        return Controller({
+            framework: framework,
+            spendingConditionRegistry: spendingConditionRegistry,
+            outputGuardHandlerRegistry: outputGuardHandlerRegistry
+        });
+    }
+
+    function run(
+        Controller memory controller,
+        PaymentExitDataModel.InFlightExitMap storage inFlightExitMap,
+        PaymentInFlightExitRouterArgs.ChallengeOutputSpent memory args
+    )
+        public
+    {
+        uint192 exitId = ExitId.getInFlightExitId(args.inFlightTx);
+        PaymentExitDataModel.InFlightExit storage ife = inFlightExitMap.exits[exitId];
+        require(ife.exitStartTimestamp != 0, "In-flight exit doesn't exists");
+
+        UtxoPosLib.UtxoPos memory utxoPos = UtxoPosLib.UtxoPos(args.outputUtxoPos);
+        uint16 outputIndex = UtxoPosLib.outputIndex(utxoPos);
+        require(
+            PaymentInFlightExitModelUtils.isOutputPiggybacked(ife, outputIndex),
+            "Output is not piggybacked"
+        );
+
+        verifyInFlightTransactionIncludedInPlasma(controller, args);
+        verifyChallengingAndInFlightTransactionAreDifferent(args);
+        verifyOutputType(controller, args);
+        verifyChallengingTransactionSpendsOutput(controller, args, ife);
+
+        PaymentInFlightExitModelUtils.cancelOutputPiggyback(ife, outputIndex);
+
+        //pay bond to challenger
+        msg.sender.transfer(PIGGYBACK_BOND);
```

Comment:
I know we have a bond mechanism coded. How can we use, to get the right bond here? @kevsul